CMake fails without Git

CMake fails if Git is not installed on the host system. 

There is no check if Git is present in the system before running Git to find out this project version from tags in Benchmark. See cmake/GetGitVersion.cmake.

Running CMake without Git yields:
```
CMake Error at cmake/GetGitVersion.cmake:28 (string):
  string sub-command STRIP requires two arguments.
Call Stack (most recent call first):
  CMakeLists.txt:11 (get_git_version)


CMake Error at cmake/GetGitVersion.cmake:29 (string):
  string sub-command REGEX, mode REPLACE needs at least 6 arguments total to
  command.
Call Stack (most recent call first):
  CMakeLists.txt:11 (get_git_version)


-- git Version: 
CMake Error at CMakeLists.txt:14 (string):
  string sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.


-- Version: 
CMake Error at CMakeLists.txt:19 (string):
  string sub-command SUBSTRING requires four arguments.
```

Im using CMake 2.8.11 (also tested CMake 3.0.2) on Linux and Windows.

I can submit a pull request, but before I want to check with you guys that this is an issue.

--- a/AUTHORS
+++ b/AUTHORS
@@ -24,3 +24,4 @@ Oleksandr Sochka <sasha.sochka@gmail.com>
 Paul Redmond <paul.redmond@gmail.com>
 Shuo Chen <chenshuo@chenshuo.com>
 Yusuke Suzuki <utatane.tea@gmail.com>
+Dirac Research 

--- a/CONTRIBUTORS
+++ b/CONTRIBUTORS
@@ -40,3 +40,4 @@ Paul Redmond <paul.redmond@gmail.com>
 Pierre Phaneuf <pphaneuf@google.com>
 Shuo Chen <chenshuo@chenshuo.com>
 Yusuke Suzuki <utatane.tea@gmail.com>
+Tobias Ulvgård <tobias.ulvgard@dirac.se>

--- a/cmake/GetGitVersion.cmake
+++ b/cmake/GetGitVersion.cmake
@@ -10,7 +10,8 @@
 # include(GetGitVersion)
 # get_git_version(GIT_VERSION)
 #
-# Requires CMake 2.6+
+# Requires CMake 2.8.11+
+find_package(Git)
 
 if(__get_git_version)
   return()
@@ -18,16 +19,21 @@ endif()
 set(__get_git_version INCLUDED)
 
 function(get_git_version var)
-  execute_process(COMMAND git describe --match "v[0-9]*.[0-9]*.[0-9]*" --abbrev=8
-      RESULT_VARIABLE status
-      OUTPUT_VARIABLE GIT_VERSION
-      ERROR_QUIET)
-  if(${status})
-      set(GIT_VERSION "v0.0.0")
+  if(GIT_EXECUTABLE)
+      execute_process(COMMAND git describe --match "v[0-9]*.[0-9]*.[0-9]*" --abbrev=8
+          RESULT_VARIABLE status
+          OUTPUT_VARIABLE GIT_VERSION
+          ERROR_QUIET)
+      if(${status})
+          set(GIT_VERSION "v0.0.0")
+      else()
+          string(STRIP ${GIT_VERSION} GIT_VERSION)
+          string(REGEX REPLACE "-[0-9]+-g" "-" GIT_VERSION ${GIT_VERSION})
+      endif()
   else()
-      string(STRIP ${GIT_VERSION} GIT_VERSION)
-      string(REGEX REPLACE "-[0-9]+-g" "-" GIT_VERSION ${GIT_VERSION})
+      set(GIT_VERSION "v0.0.0")
   endif()
+
 
   # Work out if the repository is dirty
   execute_process(COMMAND git update-index -q --refresh

